packages/enzyme-adapter-react-16.1/src/ReactSixteenOneAdapter.js

```diff
@@ -327,6 +342,16 @@ class ReactSixteenOneAdapter extends EnzymeAdapter {
             : elementToTree(output),
         };
       },
+      simulateError(nodeHierarchy, rootNode, error) {
+        simulateError(
```

Comment:
Why not check the `nodeHierarchy` for error boundaries in a shallow renderer? Should this scenario require a mount renderer?

```js
// ErrorBoundary is a component with a componentDidCatch method that renders null
// if an error is encountered.
function MyComponent({children}) {
    return (
        <div>
            <h3>My component is cool!</h3>
            <ErrorBoundary>
                {children}
            </ErrorBoundary>
        </div>
    );
}

const BadChild = () => null;

const wrapper = shallow(
    <MyComponent>
        <BadChild />
    </MyComponent>
);
wrapper.find(BadChild).simulateError(new Error('That was bad'));

expect(wrapper.find('h3')).toExist();
```

Reply:
In a shallow render, there can't ever be any other error boundaries, since only the root node is actually rendered.

Reply:
Sure, but if `simulateError` is called on an element in shallow render tree, it seems reasonable to render any error boundaries found above that element. This actually seems to work:

```js
const wrapper = shallow(
    <MyComponent>
        <BadChild />
    </MyComponent>
);
const errorBoundary = wrapper.find('ErrorBoundary').shallow();
errorBoundary.find(BadChild).simulateError(new Error('That was bad'));
expect(wrapper.find('h3')).toExist();
```

...but it doesn't really feel intuitive. I can see people trying my previous example and being confused when it doesn't work, especially since I did just that :\

Reply:
It's not reasonable or possible, because of how shallow rendering works. In shallow rendering, everything that's not the root node is treated as if it's a div - ie, as if the component implementation is `({ children }) => children`. Thus, there's no `componentDidCatch` anywhere, except the top.